### PR TITLE
Feature/progress bar circle add size caps and outer stroke

### PR DESCRIPTION
**Pull request by IgnacioND**
## Category/Component

`❓ Ask`

**TASK**: [MTR-71274](https://jira.ets.mpi-internal.com/browse/MTR-71274)

### Description, Motivation and Context
- Add props to Progress Bar component.
- New XL size.
- OuterStroke to have a progress line bigger than the main one.
- Add the possibility to change the line caps to Round type.
- Update tests and classes

### Types of changes

- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [X] 🧪 Test
- [ ] 🧠 Refactor
- [X] 💄 Styles
- [ ] 🛠️ Tool

### Screenshots - Animations
image

### CODE REVIEW

File: components/atom/progressBar/src/ProgressBarCircle/index.scss [hunk 1]
```diff
@@ -27,6 +27,14 @@ $base-class-circle: '#{$base-class}Circle';
     height: $h-atom-circle-progress-bar-circle-large;
     width: $w-atom-circle-progress-bar-circle-large;
   }
+  &--xl {
+    align-items: center;
+    display: flex;
+    justify-content: center;
+    font-size: $fz-atom-circle-progress-bar-text-xl--inner;
```
Comment (xmurcia) [inline components/atom/progressBar/src/ProgressBarCircle/index.scss]: Is that naming with double dash correct?

**Review by xmurcia — APPROVED** [review]

File: components/atom/progressBar/src/ProgressBarCircle/Circle/index.js [hunk 2]
```diff
@@ -88,6 +90,10 @@ Circle.propTypes = {
   strokeWidth: PropTypes.node.isRequired,
   /** boolean to activate/desactivate animations */
   withAnimation: PropTypes.bool,
+  /** When progress stroke is bigger than main one, it would be double in width  */
+  outerStrokeWidth: PropTypes.bool,
+  /** The shape of the end of line, it can be "round" or "square" */
+  strokeLineCaps: PropTypes.string.isRequired,
```
Comment (jordevo) [inline components/atom/progressBar/src/ProgressBarCircle/Circle/index.js]: `strokeLinecap` inside the component but `strokeLineCaps` as the name of the prop? any reason for that?

Comment (jordevo) [inline components/atom/progressBar/src/ProgressBarCircle/Circle/index.js]: being it a new `prop`, isn't making it `required` sort of a _breaking_ change?

File: components/atom/progressBar/src/ProgressBarCircle/index.scss (hunk 1 above)
Comment (jordevo) [inline components/atom/progressBar/src/ProgressBarCircle/index.scss]:
```suggestion
    font-size: $fz-atom-circle-progress-bar-text--xl;
```

perhaps?

File: components/atom/progressBar/src/settings.js [hunk 3]
```diff
@@ -7,11 +7,17 @@ export const TYPES = {
 }
 
 export const SIZES = {
+  XL: 'xl',
```
Comment (jordevo) [inline components/atom/progressBar/src/settings.js]:
```suggestion
  EXTRA_LARGE: 'extraLarge',
```

in order to be consistent with the existing naming?

File: components/atom/progressBar/src/ProgressBarCircle/Circle/index.js [hunk 4]
```diff
@@ -88,6 +90,10 @@ Circle.propTypes = {
   strokeWidth: PropTypes.node.isRequired,
   /** boolean to activate/desactivate animations */
   withAnimation: PropTypes.bool,
+  /** When progress stroke is bigger than main one, it would be double in width  */
+  outerStrokeWidth: PropTypes.bool,
```
Comment (jordevo) [inline components/atom/progressBar/src/ProgressBarCircle/Circle/index.js]: a boolean prop should have a boolean-ish name IMHO...

how about `hasOuterStrokeDoubleWidth`?

File: components/atom/progressBar/src/ProgressBarCircle/index.js [hunk 5]
```diff
@@ -67,12 +71,21 @@ ProgressBarCircle.propTypes = {
 
   /** Hide the indicator */
   hideIndicator: PropTypes.bool,
+
+  /** The shape of the end of line, it can be "round" or "square" */
+  strokeLineCap: PropTypes.oneOf(Object.values(LINE_CAPS)),
+
+  /** When progress stroke is bigger than main one, it would be double in width  */
+  outerStrokeWidth: PropTypes.bool,
```
Comment (jordevo) [inline components/atom/progressBar/src/ProgressBarCircle/index.js]: same as above https://github.com/SUI-Components/sui-components/pull/2578/files#r1236687248

**Review by jordevo — APPROVED** [review]

**Review by davidmartin84 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**